server/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java

```diff
@@ -101,9 +106,15 @@ private static SysSnapshot toSysSnapshot(Repository repository, SnapshotId snaps
     }
 
     private static CompletableFuture<SysSnapshot> createSysSnapshot(Repository repository, SnapshotId snapshotId) {
-        return repository.getSnapshotInfo(snapshotId)
-            .thenApply(snapshotInfo -> toSysSnapshot(repository, snapshotId, snapshotInfo))
-            .exceptionally(t -> {
+        return repository.getSnapshotGlobalMetadata(snapshotId).thenCombine(
+            repository.getSnapshotInfo(snapshotId),
+            (metadata, snapshotInfo) -> {
+                List<String> partedTables = new ArrayList<>();
+                for (var template : metadata.templates().values()) {
+                    partedTables.add(RelationName.fqnFromIndexName(template.value.getName()));
```

Comment:
maybe we can skip the `fqnFromIndexName` here and use it in the `tables()` of `SysSnaphot` to be in sync with concrete indices, wdyt?
@BaurzhanSakhariev @mfussenegger

Reply:
it's fine to me to leave it as is. I think in a long run we anyway want to operate with CrateDB specific terms (tables and not indices)